vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/steps/integration/RedirectValidationSteps.java

```diff
@@ -0,0 +1,184 @@
+    private final ISoftAssert softAssert;
+    private final IAttachmentPublisher attachmentPublisher;
+
+    public RedirectValidationSteps(HttpRedirectsProvider redirectsProvider, ISoftAssert softAssert,
+            IAttachmentPublisher attachmentPublisher)
+    {
+        this.redirectsProvider = redirectsProvider;
+        this.softAssert = softAssert;
+        this.attachmentPublisher = attachmentPublisher;
+    }
+
+    /**
+     * Check that all URLs redirect to proper pages with correct redirects quantity
+     * <p>Validation fails if either actual final URL or actual number of URLs not match the defined values</p>
+     * <b>Example:</b>
+     * <table border="1" style="width:70%">
+     * <caption>A table of parameters</caption>
+     * <thead>
+     * <tr>
+     * <td><b>startUrl</b></td>
+     * <td><b>endUrl</b></td>
+     * <td><b>redirectsNumber</b></td>
+     * </tr>
+     * </thead>
+     * <tbody>
+     * <tr>
+     * <td>https://example.com/redirect</td>
+     * <td>https://example.com/get-response</td>
+     * <td>1</td>
+     * </tr>
+     * </tbody>
+     * </table>
+     * @param expectedRedirects Table of parameters where: <br>
+     * <ul>
+     * <li><b>startUrl</b> - The url from which redirection begins</li>
+     * <li><b>endUrl</b> - The expected url to which redirected</li>
+     * <li><b>redirectsNumber</b> - The expected quantity redirects between startUrl and endUrl (optional)</li>
+     * </ul>
+     */
+    @Then("there are redirects with parameters: $expectedRedirects")
+    public void validateRedirects(List<ExpectedRedirect> expectedRedirects)
+    {
+        for (ExpectedRedirect redirect : expectedRedirects)
+        {
+            RedirectValidationState validationState = new RedirectValidationState(redirect);
+            try
+            {
+                checkRedirect(validationState, redirect);
+                performPostCheckActions(validationState);
+            }
+            catch (AssertionError e)
+            {
+                performPostCheckActions(validationState);
+                throw e;
+            }
+        }
+    }
+
+    private void performPostCheckActions(RedirectValidationState state)
+    {
+        attachmentPublisher.publishAttachment(
+                "/org/vividus/steps/integration/redirect-validation-test-report.ftl",
+                Map.of("result", state),
+                String.format("Redirects validation test report from '%s' to '%s'",
+                        state.getExpectedRedirect().getStartUrl(), state.getExpectedRedirect().getEndUrl()));
+    }
+
+    private void checkRedirect(RedirectValidationState validationState, ExpectedRedirect expectedRedirect)
+    {
+        try
+        {
+            List<URI> redirects = redirectsProvider.getRedirects(expectedRedirect.getStartUrl());
+            initTestResult(validationState, redirects);
+            StringBuilder assertionMessageBuilder = new StringBuilder().append(SINGLE_QUOTE)
```

Comment:
do we really need string builder?

Reply:
its a bit hard to read because the message building login is spread across big method, probably it makes sense to use just string format

Reply:
I think in this case the format doesn't quite suit for us
We have 4 different assertion message options with different number of string parts and this string parts have completely different meanings